### ISSUE TITLE
Allow admins to forward reports to other instances or relays

### ISSUE THREAD

**Issue by trwnh** [tootsuite/mastodon#15526]
### Pitch

https://github.com/tootsuite/mastodon/issues/11510#issuecomment-518908011

Forward reports to other instances or relays, using Flag activities.

### Motivation

Currently, when reporting remote accounts, users can choose to forward a report to the remote instance, but the report can only ever exist on those two instances (the origin and the recipient). Bad actors who will not be moderated on their own instances therefore have many chances to harm other users on many different instances.

Currently, the workaround to this is for users to make their own block recommendations, to their followers or perhaps to a hashtag. While this kind of emergent behavior shows a need for better moderation discoverability, it also creates a situation where each *user* has to constantly either keep up with the never-ending stream of bad actors. Furthermore, this diligence is duplicated for every single user.

If it can be said that users trust their moderation team to make decisions for them, then this duplicate work can instead be delegated to the moderation team. Rather than having vulnerable users forced to bear the brunt of moderation, I propose letting the moderation team handle it for them, because that is what a moderation team is supposed to do.

The easiest way to do this is to simply allow Flag activities to be forwarded to arbitrary instances. This would take no additional work to support incoming Flag activities, as they are already supported; perhaps the forwarding can even be automatic in the case of relays.

**Comment by ThisIsMissEm** [comment]
@trwnh would you consider this done now with the report UX changing to include mentioned servers? (Though I note it doesn't show up consistently: 
<img width="486" alt="Screenshot 2024-02-20 at 01 12 14" src="https://github.com/mastodon/mastodon/assets/30827/45a7f11f-c219-4910-822f-d85cf0c55899">

(That's just a random status I pulled up to screenshot the UI)

**Comment by trwnh** [comment]
@ThisIsMissEm no, this issue is requesting that admins be able to forward arbitrary reports, and to be able to forward them to arbitrary other mod teams. the feature you describe is for users only, and is limited to involved domains

**Comment by ThisIsMissEm** [comment]
> looks like a partial duplicate of this issue

@trwnh perhaps we can refine this issue to remove the duplication? i.e., if I'm understanding correctly, you want a way to forward report xyz to instance abc, even though the report doesn't concern instance abc?

**Comment by trwnh** [comment]
it's more like #24681 is conceptually similar to #18518, where the former *allows* admins to forward reports in addition to the user, and the latter *requires* admins to forward reports perhaps on suggestion by the user.

then, on top of that, this issue requests that the admin UI for such a feature allow arbitrary recipients. basically, it presupposes some kind of forwarding in the hands of admins as described in #11510, #24681, #18518

**Comment by ThisIsMissEm** [comment]
I'm not sure arbitrary forwarding is a good idea, as it could be used as a harassment / spam vector (though you could certainly disable reports from those instances)

**Comment by trwnh** [comment]
> you could certainly disable reports from those instances

https://github.com/mastodon/mastodon/issues/7608#issuecomment-391503658

**Comment by ThisIsMissEm** [comment]
@trwnh out of curiosity, would this create new Flag activities, or would it be more a `Announce(Flag)`?

**Comment by trwnh** [comment]
This doesn't require any changes, so it wouldn't create a new activity nor would it announce an existing one -- it would just set arbitrary addressing recipients. So you could have `userA@site1.example` report `userB@site2.example`, but then the mods of `site1.example` decide to pass on the report to `site3.example` or `relay.example` arbitrarily. The `Flag` activity would be like any other normal one, just with those recipients in `to` or `cc`:

```yaml
actor: <site1.example/actor>
type: Flag
object: <site2.example/users/userB>
to: [<site2.example/actor>, <site3.example/actor>, <relay.example>]
```

**Comment by ThisIsMissEm** [comment]
Right, but wouldn't this be most used in the case of asking another moderation team for help on a report (e.g., due to language barriers), where the Flag isn't actually owned by the instance forwarding it

**Comment by trwnh** [comment]
I feel like that's a separate feature request -- this one is not about asking for help with how to deal with a report locally, but instead is about reducing moderation load/burden for instances that want to share that load.

**Comment by ThisIsMissEm** [comment]
oooh, okay. Given that moderation actions only happen on a single instance, and there's no federation of updates, I'm not sure that this makes sense at all?

At least, it's not something that could be implemented any time soon.

**Comment by trwnh** [comment]
It's already done for all mentioned users' servers as you point out in https://github.com/mastodon/mastodon/issues/15526#issuecomment-1953300940 -- the difference between this vs. what I'm requesting is that I'm requesting that the *mod team* gets to decide to forward (as described in #11510 / #24681 / #18518 mentioned above), and specifically that the mod team gets to forward to *arbitrary recipients*. So each server's mod team gets to make their own moderation action, but also, different servers' mod teams get to share information in-band.

**Comment by ThisIsMissEm** [comment]
You could only change to/cc based on if your instance is the owner of the report; if the report is from a remote instance, the Flag cannot be modified by your instance (since you don't own it), hence suggesting using `Announce(Flag)` for this.

**Comment by trwnh** [comment]
this feature request is specifically for reports generated by your instance

if you were to have a separate request for forwarding reports that were
themselves forwarded to you, then this could be achieved through any of the
existing forwarding or relaying mechanisms (POST to inbox with either a
relayed signature, or wrapped in an Announce, or making it dereferencible
with the expectation that later recipients will fetch it for verification)
